solver/src/driver.rs

```diff
@@ -327,9 +328,13 @@ impl Driver {
         futures::stream::iter(settlements)
             .filter_map(|(solver, settlement)| async {
                 let surplus = settlement.total_surplus(prices);
-                // Because of a potential fee discount, the solver fees may by themselves not be sufficient to make a solution economically viable (leading to a negative objective value)
-                // We therefore reverse apply the fee discount to simulate unsubsidized fees for ranking.
-                let unsubsidized_solver_fees = settlement.total_fees(prices) / BigRational::from_float(self.fee_factor).expect("Discount factor is not a rational");
+                let mut unsubsidized_solver_fees = settlement.total_unsubsidized_fees(prices);
+                // Old orders don't have unsubsidized fees set, so we use the old logic
+                // of restoring the unsubsidized fee from the subsidized one by reversing the fee factor.
+                // This logic will be removed in the future.
+                if unsubsidized_solver_fees.is_zero() {
+                    unsubsidized_solver_fees = settlement.total_fees(prices) / BigRational::from_float(self.fee_factor).expect("Discount factor is not a rational");
+                }
```

Comment:
This wouldn't be correct when one settlement contains some orders with the new field set and some without, right? But I don't think it's a problem since that should be a short time period.

Reply:
Yeah you're right. I'll add a check for this.